lib/azureServiceClient.ts

```diff
@@ -115,67 +87,81 @@ export class AzureServiceClient extends msRest.ServiceClient {
    * @returns {Promise<msRest.HttpOperationResponse>} result - The final response after polling is complete.
    */
   async getLongRunningOperationResult(resultOfInitialRequest: msRest.HttpOperationResponse, options?: msRest.RequestOptionsBase): Promise<msRest.HttpOperationResponse> {
-    const self = this;
     const initialRequestMethod: string = resultOfInitialRequest.request.method as msRest.HttpMethods;
 
-    if (self.checkResponseStatusCodeFailed(resultOfInitialRequest)) {
-      return Promise.reject(`Unexpected polling status code from long running operation ` +
+    if (checkResponseStatusCodeFailed(resultOfInitialRequest)) {
+      throw new Error(`Unexpected polling status code from long running operation ` +
```

Comment:
Is it okay to throw from a method that returns a Promise? I thought it was okay with async methods, but I ended up being wrong according to warning messages I kept getting on the command line.

Reply:
If you never `.catch` it node will be mad at you. But it's ok to throw in a `.then()` callback. See https://stackoverflow.com/questions/33445415/javascript-promises-reject-vs-throw#33446005